### PR TITLE
Allow .setup to update the merchantID and session token

### PR DESCRIPTION
**Pull request by djoksimo**
## What

- Make some methods `static` instead of `class` functions
- Remove dangerous references to `ForageSDK.shared.` that cause the logger to use the wrong environment
- Update sample app to cover the `.setup()` updates.

## Test Plan

- [x] Added unit test for `ForageSDK.setup`
- [x] Try calling `.setup` and different stages of the render lifecycle
- [x] Make sure logs appear where they're supposed to appear
- [ ] Make sure feature flags load as expected

## How

Can be released as-is. A clean up + refactoring ticket will be created.

### CODE REVIEW

File: Sample/SampleForageSDK/ViewController.swift [hunk 1]
```diff
@@ -30,11 +30,15 @@ class ViewController: UIViewController {
...
+        /// the config values later on in the ``CardNumberViewController.viewDidAppear`` method
         ForageSDK.setup(
             ForageSDK.Config(
-                merchantID: merchantID,
-                sessionToken: sessionToken
+                merchantID: "invalid mid/...",
+                // intentionally omitted environment prefix from this sessionToken
+                sessionToken: "invalid session Token with no environment prefix"
             )
         )
```
Comment (djoksimo) [inline Sample/SampleForageSDK/ViewController.swift:43]: We can also add a test case in the Sample app that simply goes back and re-submits new session token + merchant ID, but that will take some time to introduce

So in the meantime, let's keep this code - unless @dleis612 objects

File: Sources/ForageSDK/ForageSDK.swift [hunk 2]
```diff
@@ -19,6 +19,9 @@ public class ForageSDK {
     public var environment: Environment = .sandbox
     // Don't update! Only updated when releasing.
     public static let version = "4.4.7"
+    
+    /// Until we migrate away from the singleton pattern in `ForageSDK`
+    /// ...Reference `shared` with caution! It can have dangerous side effects
```
Comment (devinmorgan) [inline Sources/ForageSDK/ForageSDK.swift:24]: Ooooo didn't realize we were considering this - very exciting!

File: Tests/ForageSDKTests/ForageSDKTests.swift [hunk 3]
```diff
@@ -34,6 +34,19 @@ final class ForageSDKTests: XCTestCase {
 
         XCTAssertEqual(MockForageSDK.shared.sessionToken, "dev_authToken123")
     }
+    
+    func testSetup_shouldUpdateSessionTokenAndMerchantID() {
```
Comment (dleis612) [inline Tests/ForageSDKTests/ForageSDKTests.swift:38]: TODO: Add a test of setting the config, reading `ForageSDK.shared.environment`, then setting the config again.

**Comment by djoksimo** [comment]
Closing this PoC in favor of Danny's upcoming changes

**Pull request closed without merging**